src/main/webapp/app/assessment/athena.service.ts

```diff
@@ -13,28 +14,52 @@ export class AthenaService {
         private profileService: ProfileService,
     ) {}
 
+    public isEnabled(): Observable<boolean> {
+        return this.profileService.getProfileInfo().pipe(switchMap((profileInfo) => of(profileInfo.activeProfiles.includes('athena'))));
+    }
+
     /**
      * Get feedback suggestions for the given submission from Athena
      *
-     * @param exerciseId the id of the exercise
+     * @param exercise
      * @param submissionId the id of the submission
+     * @return observable that emits the feedback suggestions
      */
-    getFeedbackSuggestions(exerciseId: number, submissionId: number): Observable<TextBlockRef[]> {
-        return this.profileService.getProfileInfo().pipe(
-            switchMap((profileInfo) => {
-                if (!profileInfo.activeProfiles.includes('athena')) {
-                    return of([] as TextBlockRef[]);
+    private getFeedbackSuggestions<T>(exercise: Exercise, submissionId: number): Observable<T[]> {
+        if (!exercise.feedbackSuggestionsEnabled) {
+            return of([]);
+        }
+        return this.isEnabled().pipe(
+            switchMap((isAthenaEnabled) => {
+                if (!isAthenaEnabled) {
+                    return of([] as T[]);
                 }
                 return this.http
-                    .get<TextBlockRef[]>(`${this.resourceUrl}/exercises/${exerciseId}/submissions/${submissionId}/feedback-suggestions`, { observe: 'response' })
-                    .pipe(switchMap((res: HttpResponse<TextBlockRef[]>) => of(res.body!)))
-                    .pipe(
-                        switchMap((feedbackSuggestions) => {
-                            // Make real TextBlockRef objects out of the plain objects
-                            return of(feedbackSuggestions.map((feedbackSuggestion) => new TextBlockRef(feedbackSuggestion.block!, feedbackSuggestion.feedback)));
```

Comment:
Note for reviewers: This is now also handled in `text-submission-assessment.component.ts`, `loadFeedbackSuggestions`